cart-service/src/app.ts

```diff
@@ -16,6 +16,26 @@ app.disable('x-powered-by');
 app.use(bodyParser.json());
 
 app.post('/cart-service', async (req: Request, res: Response) => {
+
+  //check if the request has a basic auth header
+  if (!req.headers.authorization) {
+    setErrorResponse(res, 403, 'Forbidden');
+    return;
+  }
+
+  //check if the request header has the correct basic auth password ane return 403 if it does not
+
+  const basicAuthPassword = process.env.CONNECTOR_BASIC_AUTH_PASSWORD;
```

Comment:
This can come from the `configuration` variable:

```suggestion
  const basicAuthPassword = configuration.connectorBasicAuthPassword;
```

This should then mean you don't need to set the `CONNECTOR_BASIC_AUTH_PASSWORD` environment variable in the `build.yml` file.